[WIP - SAF 282] Implement self reporting screens

zhaccc opened the pull request:
#### Description:

This PR fixes comments on the existing PR that was open a while ago but abandoned. It also implements new design and does some tweaks to existing code.

#### Screenshots:

![self-assessment](https://user-images.githubusercontent.com/2924479/84441092-285ba800-ac09-11ea-8e68-ad2a64ce7faa.gif)

qingqingliu committed:
add support for display location from waypoints api

qingqingliu committed:
fix lint and test errors

qingqingliu committed:
add auto format support

qingqingliu committed:
undo eslintrc change

qingqingliu committed:
Merge pull request #22 from safe-paths-contrib/qingqing/waypoints_map

Add support for getting location from waypoints API and display it on map

qingqingliu committed:
Merge branch 'develop' of github.com:tripleblindmarket/covid-safe-paths into self-reporting

qingqingliu committed:
pull latest change from dev

qingqingliu committed:
Merge pull request #23 from safe-paths-contrib/qingqing/pickup_dev

Pick up latest changes from origin development branch

qingqingliu committed:
upload waypoints data

kujenga committed:
github/workflows: temporarily build on self-reporting branch

This facilitates development on our fork while we are preparing a
unified contribution to the upstream.

qingqingliu committed:
Merge pull request #25 from safe-paths-contrib/at/gh-actions-build

Enabling GitHub actions build on forked branch

qingqingliu committed:
remove the usage of actual location data for upload

qingqingliu committed:
Merge pull request #24 from safe-paths-contrib/qingqing/axios

upload waypoints data

qingqingliu committed:
initial check in for Captcha

qingqingliu committed:
remove unneeded patching

qingqingliu committed:
Merge pull request #33 from safe-paths-contrib/qingqing/captcha

initial check in for Captcha

Commit:
<Assessment />

qingqingliu committed:
add assessment captcha screen

qingqingliu committed:
Merge branch 'self-reporting' of github.com:safe-paths-contrib/covid-safe-paths into survey

# Conflicts:
#	app/Entry.js

qingqingliu committed:
adjust height for captcha and introduce scroll view

qingqingliu committed:
hook in survey api

Commit:
Cleanup

Commit:
Update AssessmentQuestion.js

qingqingliu committed:
introduce loading indicator to captcha

qingqingliu committed:
Merge branch 'survey' of github.com:safe-paths-contrib/covid-safe-paths into survey

Commit:
Pepare response payload

Commit:
Merge branch 'survey' of github.com:safe-paths-contrib/covid-safe-paths into survey

* 'survey' of github.com:safe-paths-contrib/covid-safe-paths:
  introduce loading indicator to captcha

qingqingliu committed:
add support for survey upload with captcha header

Commit:
Survey

Commit:
Typo

Commit:
Cleanup

Commit:
Cleanup

Commit:
Cleanup

Commit:
Merge remote-tracking branch 'official/develop' into survey

* official/develop: (110 commits)
  Consolidate to a single theme aware Button component (#767)
  Fixed incorrect pull command. (#769)
  Update README, etc for tripleblind -> Path-Check (#768)
  HTML EULA translation workflow (#710)
  Allowing setup on macOS/Darwin for Android only. (#766)
  Add Danish to the list of languages. (#765)
  Fixed the aspect ration on the icon and add links (#763)
  Build time feature flags (Phase 1) (#694)
  Extract theme aware <IconButton /> component (#756)
  Latest i18n pull (#755)
  Add language contribution info to contributing guide (#754)
  Recommend PR best practices in PR template (#750)
  Add eslint react-hooks checks (#751)
  iOS: Added local notification in terminate function. (#521)
  Update dev README (#705)
  Better error logging in Google import (#703)
  Remove rebasing from suggestions in CONTRIBUTING.md (#724)
  Translation refresh (#708)
  updated issue template (#729)
  Set up location after skiping the steps (#721)
  ...

Commit:
Remove WayPoints

Commit:
Merge fixes

Commit:
Fix label key

Commit:
Update build.yml

Commit:
Delete customCircle.js

Commit:
Merge fixes

Commit:
Merge fixes

Commit:
Update Settings.spec.js.snap

Commit:
Update App.spec.js.snap

Commit:
Remove axios

Commit:
Feature flag

Commit:
Lint

Commit:
Component cleanup

Commit:
Import images

Commit:
Delete settings.json

Commit:
Move enableScreens

Commit:
Constants

Commit:
Create assessment folder for images

Commit:
Named export

Commit:
Merge remote-tracking branch 'official/develop' into self-assessment

* official/develop:
  Fix EULA modal close icon color (#792)
  feat(i18n) Language updates, May 10 (#790)
  feat(GPS) Store Location Data in an encrypted Realm DB (#788)
  Enable location status checks in testing build (#787)
  fix(Onboarding) Fix EULA links to open in device browser (#783)
  chore(Build) Fix debug build error with apache commons-io lib (#775, SAF-210)
  chore(Testing) snapshot tests for LocationTracking,  main/* (#785)
  Only render mayo clinic link when language is english (#782)
  fix(UX) Fix pulse alignment on Android devices with “notch” (#771)
  fix(Location) Improve location "tracking" status checks (#749)
  Fix text crop in About screen (#774)
  [iOS] add missing languages (#773)

Commit:
Cleanup

Commit:
Tests

Commit:
Merge branch 'self-assessment' of github.com:safe-paths-contrib/covid-safe-paths into SAF-408/implement-self-reporting-screens

Commit:
Fix comments and add fonts

Commit:
Merge branch 'develop' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

Commit:
Remove feature flag from self assessment

Commit:
Remove select gender and share location data screens

Commit:
Merge branch 'develop' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

jeffstolz committed:
wip

Commit:
Merge branch 'develop' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

Commit:
Handle single value questions in multi question type environment

Commit:
Merge branch 'develop' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

Commit:
Merge branch 'js-self-assessment-design-updates' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

Commit:
Update test snapshots

jeffstolz committed:
Update start and end screens, buttons, pickers, headers, backgrounds.
Remove illustrations and refactor existing code.

jeffstolz committed:
Update start and end screens, buttons, pickers, headers, backgrounds.
Remove illustrations and refactor existing code.

Commit:
Fix captcha background color

Commit:
Merge branch 'js-self-assessment-design' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

Commit:
Merge branch 'js-self-assessment-design' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

tparesi commented on app/views/assessment/survey.en.json:
```diff
@@ -0,0 +1,399 @@
+{
```
Will this be translated properly or does it have to be moved under the locale folder (specifically en.json)?

tparesi reviewed (approved):
Really hard to get a feel for all the changes, but I think it all looks good. I know this is a draft at this point, but I don't see much leftover.

Commit:
Merge branch 'develop' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

Commit:
Merge branch 'develop' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

Commit:
Merge branch 'develop' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

Commit:
Fix removed assessment screen from navigation stack

Commit:
Merge branch 'develop' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

Commit:
Update yarn.lock

Commit:
Prettier fix

Commit:
Rename svg files to ts

Commit:
Fix lint errors

Commit:
Fix prettier errors

Commit:
Fix tests

Commit:
Refactor folder structure and file naming within views/assessment

This commit:
To clean up the code a bit, removed Assessment prefix from all files inside of the assessment folder.
Remove unused captcha files.
Switched to named exports (instead of default)

Commit:
Add test for assessment Share end screen

Commit:
Merge branch 'develop' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

Commit:
Fix prettier and tests

Commit:
Remove unused fonts

johnschoeman committed:
Allow users to select day on en calendar

Why:
We would like for users to be able to select various days on the
calendar and see any detail information about their exposure history on
that day.

This commit:
Introduces the above functionality. Currently much of the existing
calendar views are tightly coupled to the gps exposure calculations and
determinations. This commit introduces a parallel set of calendar
components to be rendered on the bt version of the app. The notion of an
ExposureHistory was introduced and the determination of data was kept
separate from the view rendering logic. Ideally we can introduce the
same separation of concerns on the gps side in a future commit and
remove redundant components. We did not do that in this pr for sake of
time and to keep the commits smaller.

Note that we introduced some fake data to mock out the exposure history
until we can get the actual data wired up.

johnschoeman committed:
Show next steps screen for affected users

Why:
When a user has a potential exposure, we would like to indicate to them
what the next steps are.

This commit:
Introduces a next steps screen, and wires up the navigation to that
screen from the ExposureHistory screen. Note that the content is
currently hard coded and will need to be replaces with HA specific
content in a future commit.

mattThousand commented on app/locales/en.json:
```diff
@@ -1,4 +1,40 @@
+    "share_title": "Publish anonymized data",
+    "share_description": "During the interview with <HA name> your interviewer can remove locations you would like to keep private, such as homes and apartments. They can also add any additional locations you believe should be included.\n\n<HA name> may share the GPS coordinates from your location history, including date, time, and duration of your visits to let others in your community know if t ...
+    "share_screening_title": "Anonymized Screening Data",
+    "share_screening_description": "Share anoymized answers to screening questions with your local healthcare authority.",
```
~anoymized~ anonymized

Commit:
Merge branch 'develop' into SAF-408/implement-self-reporting-screens

jeffstolz committed:
remove old images, clean up broken styling (#996)

Commit:
Fix merge-related crash.

Commit:
Merge branch 'jts-next-steps' into SAF-408/implement-self-reporting-screens

Commit:
Navigate to self-assessment screen from exposure history screen.

jeffstolz committed:
Wire self assessment into the tab bar, add icons

jeffstolz committed:
Update permission screen icon and copy

Commit:
Reference emptyBackgroundImage correctly.

Commit:
Reset stack on dismiss for self assessment flow.

jeffstolz committed:
Update end screen icons and backgrounds

Commit:
Merge branch 'develop' into SAF-408/implement-self-reporting-screens

jeffstolz committed:
Update snapshots

johnschoeman reviewed (approved):
THIS IS A BIG BOY.

I think theres a lot of improvements that could be made, in terms of consistency and factoring, that said, for sake of getting this long running branch in, let's merge it in and make any future code changes incrementally.

Commit:
Merge branch 'SAF-408/implement-self-reporting-screens' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

jeffstolz committed:
fix formatting errors

Commit:
Merge branch 'SAF-408/implement-self-reporting-screens' of github.com:Path-Check/covid-safe-paths into SAF-408/implement-self-reporting-screens

Commit:
Use correct gradle version.

Commit:
Fix typo/
